2_adventures/001/lenn4rd/03_raindrops_exercise/raindrops.rb

```diff
@@ -0,0 +1,25 @@
+class Fixnum
+  def prime_factor_of?(number)
+    number % self == 0
+  end
+end
+
+class Raindrops
+  PRIME_FACTORS = [3, 5, 7]
+
+  SOUNDS_FOR_FACTORS = {
+    '3' => 'Pling',
+    '5' => 'Plang',
+    '7' => 'Plong'
+  }
+  
+  def self.convert(number)
+    PRIME_FACTORS.each_with_object(sound = '') do |factor|
+      sound << SOUNDS_FOR_FACTORS[factor.to_s] if factor.prime_factor_of?(number)
+    end
+    
+    return number.to_s if sound.empty?
+    
+    sound
```

Comment:
A lot of Rubyists do stop using them, eschewing the word versions.  Until they become more proficient, and decide to use one for flow control and the other for logic statements.

The difference, besides immediate cognitive value, is the precedence for `and` and `or` are lower.  So if you have an argument to a method without parenthesis the stronger precedence will attach to the argument, and the lower precedence will attach to the result (evaluation) of the method.  Knowing this, then, then it is quickly understood, but still if that isn't at the top of your head, may make it harder to see.

It is very similar to the issue of using `do..end` or `{..}`.  The precedence is different, but in addition to that, you can use them each to communicate your code differently.  Are you using it and relying on a "side effect" or are you more interested in the result of the call in the block?

The `{..}` style lends itself to using the return naturally, the `do..end` is a little bit strange to see a `end.something`.

The point of my rambling is that the choices you make for things that can be used identically can communicate to the reader more than just what technically the statement is saying/doing.